packages/plugin-core/src/commands/LockPreview.ts

```diff
@@ -0,0 +1,40 @@
+import { PreviewProxy } from "../components/views/PreviewProxy";
+import { DENDRON_COMMANDS } from "../constants";
+import { BasicCommand } from "./base";
+
+type CommandInput = {};
+
+type CommandOpts = {};
+type CommandOutput = CommandOpts;
+
+export class LockPreviewCommand extends BasicCommand<
+  CommandOpts,
+  CommandOutput,
+  CommandInput
+> {
+  key = DENDRON_COMMANDS.LOCK_PREVIEW.key;
+  _panel: PreviewProxy | undefined;
+
+  constructor(previewPanel: PreviewProxy | undefined) {
+    super();
+    this._panel = previewPanel;
+  }
+
+  async sanityCheck() {
+    if (!this._panel || !this._panel.isVisible()) {
+      return "No preview currently open";
+    }
+    if (this._panel && this._panel.isLocked()) {
```

Comment:
can we implement this as toggling behavior? rename to toggle preview lock

Reply:
[`2ad67d2` (#3293)](https://github.com/dendronhq/dendron/pull/3293/commits/2ad67d2337df532b7348c34e12eeab5344c91c4f)
[`e827c0f` (#3293)](https://github.com/dendronhq/dendron/pull/3293/commits/e827c0f0d458712e63645f43a3bbbd17b7cf7fa9)